src/mutex.rs

```diff
@@ -311,6 +321,23 @@ mod tests {
         assert_eq!(contents, *(deserialized.lock()));
     }
 
+    #[cfg(feature = "arc_lock")]
+    #[test]
+    fn test_arc_map() {
+        use lock_api::ArcMutexGuard;
+        use std::sync::Arc;
+
+        let contents: Vec<u8> = vec![0, 1, 2];
+        let mutex = Arc::new(Mutex::new(contents));
+
+        let guard = mutex.lock_arc();
+        let mapped_guard = ArcMutexGuard::map(guard, |contents: &mut Vec<u8>| &mut contents[1]);
+        // The point of the ArcMutexGuard is that we don't borrow the mutex.
+        drop(mutex);
+
+        assert_eq!(*mapped_guard, 1);
```

Comment:
This test only shows that the happy path works. It's almost more of an example than a full test.

If desired, we could use more extensive tests to:
* Use a custom `Drop` impl to track when the value is dropped.
* Try (and catch_unwind) a mapping function that panics, see if the mutex is still usable.
* The same after running multiple mapping steps.
* Use some unsized type to show that they still work in either "core" or "mapped" position.

There's almost no upper limit to how detailed tests we could have. Just a lower limit to what we need :)